tests/binop_tests.rs

```diff
@@ -0,0 +1,76 @@
+//! Tests for binary comparisons.
+
+extern crate weld;
+
+use weld::runtime::WeldRuntimeErrno;
+
+mod common;
+use common::*;
+
+#[test]
+fn simple_binop() {
+    let code = "|| 3.1 > 4.2";
+    let ref conf = default_conf();
+
+    let ref input_data: f64 = 0.0;
+
+    let ret_value = compile_and_run(code, conf, input_data);
+    let data = ret_value.data() as *const bool;
+    let result = unsafe { *data };
+    assert!(result == false);
+}
+
+#[test]
+fn struct_cmp() {
+    let code = "|| {3.0, 2.0} < {1.0, 2.0}";
+
+    let ref conf = default_conf();
+
+    let ref input_data: f64 = 0.0;
+
+    let ret_value = compile_and_run(code, conf, input_data);
+    let data = ret_value.data() as *const bool;
+    let result = unsafe { *data };
+    assert!(result == false);
+}
+
+#[test]
+fn string_cmp() {
+    let code = "|| \"abc\" > \"def\"";
+
+    let ref conf = default_conf();
+
+    let ref input_data: f64 = 0.0;
+    
+    let ret_value = compile_and_run(code, conf, input_data);
+    let data = ret_value.data() as *const bool;
+    let result = unsafe { *data };
+
+    assert!(result == false);
+}
+
+#[test]
+fn vector_cmp() {
```

Comment:
Also add a test that compares vectors that are substrings of one another, e.g., `[1] == [1, 2, 3]`. Currently, that comparison will return `true`: I think you can add the code I put in the comments below to fix it.